rs/sns/swap/src/swap.rs

```diff
@@ -3374,128 +3376,182 @@ impl SnsNeuronRecipe {
+        {
+            let claimed_status = claimed_status.ok_or_else(|| {
+                (
+                    ConversionError::Invalid,
+                    format!(
+                        "Missing claimed_status information for neuron recipe {:?}",
+                        self
+                    ),
+                )
+            })?;
+            let claimed_status = ClaimedStatus::try_from(claimed_status).map_err(|err| {
+                (
+                    ConversionError::Invalid,
+                    format!(
+                    "Error interpreting claimed_status `{}` as ClaimedStatus for neuron recipe \
+                    {:?}: {}", claimed_status, self, err
+                ),
+                )
+            })?;
+            match claimed_status {
+                ClaimedStatus::Success => {
+                    return Err((
+                        ConversionError::Skip,
+                        format!(
+                            "Recipe {:?} was claimed in previous invocation of \
+                             claim_swap_neurons(). Skipping",
+                            self,
+                        ),
+                    ));
+                }
+                ClaimedStatus::Invalid => {
+                    // If the Recipe is marked as invalid, intervention is needed to make valid
+                    // again. As part of that intervention, the recipe must be marked
+                    // as ClaimedStatus::Pending to attempt again.
+                    return Err((
+                        ConversionError::Invalid,
+                        format!(
+                        "Recipe {:?} was invalid in a previous invocation of claim_swap_neurons(). \
+                        Skipping", self
+                    ),
+                    ));
+                }
+                // Remaining cases are tolerable. TODO: explain why.
+                ClaimedStatus::Unspecified | ClaimedStatus::Pending | ClaimedStatus::Failed => (),
+            }
+        }
+
+        let NeuronAttributes {
+            dissolve_delay_seconds,
+            memo,
+            followees,
+            ..
+        } = neuron_attributes;
+
+        let (participant, controller) = match investor {
+            Investor::Direct(DirectInvestment { buyer_principal }) => {
+                let parsed_buyer_principal = match string_to_principal(&buyer_principal) {
                     Some(p) => p,
                     // principal_str should always be parseable as a PrincipalId as that is enforced
-                    // in `refresh_buyer_tokens`. In the case of a bug due to programmer error, increment
-                    // the invalid field. This will require a manual intervention via an upgrade to correct
+                    // in `refresh_buyer_tokens`. In the case of a bug due to programmer error,
+                    // increment the invalid field. This will require a manual intervention via
```

Comment:
This is my bad because you just kept the comment, but we no longer should say "increment the invalid field" here imo because we now do that in another part of this file